### PR TITLE
update and remove integration tests

### PR DESCRIPTION
**Pull request by kalefranz**
found a legit bug!!!

@mcg1969 see test output once it's ready

### CODE REVIEW

**Comment by mcg1969** [comment]
nice!

File: conda/plan.py [hunk 1]
```diff
@@ -383,8 +383,8 @@ def add_defaults_to_specs(r, linked, specs, update=False):
             # if Python/Numpy is already linked, we add that instead of the
             # default
             log.debug('H3 %s' % name)
-            fkey = names_linked[name] + '.tar.bz2'
-            info = r.index[fkey]
+            fkey = names_linked[name]
+            info = r.index[fkey if fkey.endswith('.tar.bz2') else fkey + '.tar.bz2']
```
Comment (mcg1969) [inline conda/plan.py]: Actually, `fkey` needs to end in `.tar.bz2` later in the code. So you need to do this:
```
if not fkey.endswith('.tar.bz2'):
    fkey += '.tar.bz2'
```
or something like that

**Comment by mcg1969** [comment]
Is this a fork? I'd like to push a change

**Comment by kalefranz** [comment]
Oh yeah it's on my fork. I already fixed that though.  Is that the change you wanted?

**Comment by kalefranz** [comment]
Also note this commit: https://github.com/conda/conda/pull/2625/commits/d2d54193a1d4beee8362114fc5c4b953ffbc4e8a

**Comment by mcg1969** [comment]
No I think we need to do something different. This is what I propose: just before the construction of `names_linked` do this:
```
linked = [d if d.endswith('.tar.bz2') else d + '.tar.bz2' for d in linked]
```
then you can remove the '.tar.bz2' appending later in the code.

**Comment by mcg1969** [comment]
Actually, let me check that commit. It could be fine

File: conda/plan.py [hunk 2]
```diff
@@ -383,7 +383,9 @@ def add_defaults_to_specs(r, linked, specs, update=False):
             # if Python/Numpy is already linked, we add that instead of the
             # default
             log.debug('H3 %s' % name)
-            fkey = names_linked[name] + '.tar.bz2'
+            fkey = names_linked[name]
+            if not fkey.endswith('.tar.bz2'):
+                fkey += '.tar.bz2'
```
Comment (mcg1969) [inline conda/plan.py]: Nevermind what I said in the main conversation, this is fine.

**Comment by kalefranz** [comment]
Like this? https://github.com/conda/conda/pull/2625/commits/5520df6dbdc032c43b7669fd67d028c78d6b57e3

**Comment by mcg1969** [comment]
LGTM

**Comment by kalefranz** [comment]
Now we just need to get these integration tests passing on CI...

**Comment by mcg1969** [comment]
Yeah, I haven't been able to get them to fail here

**Comment by kalefranz** [comment]
Well that's frustrating. Does that latest one pass, with the `--clone` part?

**Pull request merged**